src/qt/walletcontroller.cpp

```diff
@@ -97,7 +98,16 @@ WalletModel* WalletController::getOrCreateWallet(std::unique_ptr<interfaces::Wal
     m_wallets.push_back(wallet_model);
 
     connect(wallet_model, &WalletModel::unload, [this, wallet_model] {
-        removeAndDeleteWallet(wallet_model);
+        // Defer removeAndDeleteWallet when no modal widget is active.
+        if (QApplication::activeModalWidget()) {
+            connect(qApp, &QApplication::focusChanged, wallet_model, [this, wallet_model]() {
```

Comment:
Just noting that the 3rd parameter `wallet_model` is used as the destination context, which means:
 - the connection is destroyed when the `wallet_model` is destroyed
 - the connection runs on the wallet model thread (GUI thread) due to `Qt::QueuedConnection`.